src/Datadog.Trace/Agent/Transports/HttpStreamRequest.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Agent.MessagePack;
+using Datadog.Trace.HttpOverStreams;
+using Datadog.Trace.HttpOverStreams.HttpContent;
+
+namespace Datadog.Trace.Agent.Transports
+{
+    internal class HttpStreamRequest : IApiRequest
+    {
+        private readonly Uri _uri;
+        private readonly DatadogHttpClient _client;
+        private readonly Stream _requestStream;
```

Comment:
Who's responsible for disposing of these Streams? Should that be this object?

Reply:
I moved it to the IApiRequest